clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java

```diff
@@ -332,6 +329,9 @@ public FetchRequestData build() {
                     iter.remove();
                     // Indicate that we no longer want to listen to this partition.
                     removed.add(topicPartition);
+                    // If we do not have this topic ID in the session, we can not use topic IDs
```

Comment:
One alternative to this approach is to only do this check when no partitions were added to the builder, but I'm not sure this added complexity is worth it. The method here is also safer.